design-toolkit/components/src/components/button/index.tsx

```diff
@@ -36,6 +36,40 @@ function ButtonProvider({ children, ...props }: ProviderProps<ButtonProps>) {
 }
 ButtonProvider.displayName = 'Button.Provider';
 
+/**
+ * Button - A versatile interactive button component with multiple variants
+ *
+ * Provides accessible button functionality with support for different visual styles,
+ * sizes, and interactive states. Includes icon support and integrates with React Aria
+ * for keyboard navigation and accessibility features.
+ *
+ * @example
+ * // Basic button
+ * <Button>Click me</Button>
+ *
+ * @example
+ * // Primary button with different sizes
+ * <Button variant="filled" size="large">Large Filled</Button>
+ * <Button variant="outline" size="small">Small Outline</Button>
+ *
+ * @example
+ * // Button with icon
+ * <Button variant="flat">
+ *   <Icon><Plus /></Icon>
+ *   Add Item
+ * </Button>
+ *
+ * @example
+ * // Icon-only button
+ * <Button variant="icon">
+ *   <Icon><Settings /></Icon>
+ * </Button>
+ *
+ * @example
+ * // Button with different colors
+ * <Button color="critical">Critical Button</Button>
+ * <Button color="serious">Delete</Button>
+ */
```

Comment:
You should add comments for LinkButton and ToggleButton too

Reply:
Done.